pkg/controllers/user/rbac/legacy_grb_cleaner.go

```diff
@@ -61,3 +81,42 @@ func (p *grbCleaner) removeStuckFinalizer(obj *v3.GlobalRoleBinding) (*v3.Global
 	}
 	return obj, nil
 }
+
+func (p *grbCleaner) removeFinalizerFromNonExistentCluster(obj *v3.GlobalRoleBinding, clusterList []*v3.Cluster) (*v3.GlobalRoleBinding, error) {
+	obj = obj.DeepCopy()
+	modified := false
+
+	md, err := meta.Accessor(obj)
+	if err != nil {
+		return obj, err
+	}
+	clusterMap := buildClusterMap(clusterList)
+
+	finalizers := md.GetFinalizers()
+	for i := len(finalizers) - 1; i >= 0; i-- {
+		f := finalizers[i]
+		if strings.HasPrefix(f, lifecycle.ScopedFinalizerKey) {
+			s := strings.Split(f, "_")
+			// if cluster was not reported, its a deleted cluster and will cause the finalizer to hang in the future
+			if _, found := clusterMap[s[1]]; !found {
+				finalizers = append(finalizers[:i], finalizers[i+1:]...)
+				modified = true
+			}
+		}
+	}
+
+	if modified {
+		md.SetFinalizers(finalizers)
+		obj, e := p.m.workload.Management.Management.GlobalRoleBindings("").Update(obj)
```

Comment:
Can just return this directly. Feels weird to return e if you know that e is going to be `IsNotFound`, and then putting that check in the caller, although `removeStuckFinalizer` follows the same pattern.

Reply:
Nice improvement 🎆